.github/workflows/test_linux_cuda.yaml

```diff
@@ -0,0 +1,66 @@
+
+name: Unit-tests on Linux GPU
+
+on:
+  pull_request:
+  push:
+    branches:
+      - nightly
+      - main
+      - release/*
+  workflow_dispatch:
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        python_version: ["3.9", "3.10"]
+        # TODO: Add more cuda versions.
+        cuda_arch_version: ["12.4"]
+      fail-fast: false
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    with:
+      runner: linux.g5.4xlarge.nvidia.gpu
+      repository: pytorch/torchcodec
+      gpu-arch-type: cuda
+      gpu-arch-version: ${{ matrix.cuda_arch_version }}
+      timeout: 120
+
+      script: |
+        nvidia-smi
+        ls -l
+        pwd
+        conda create --yes --name test
+        conda activate test
+        conda install --quiet --yes pip cmake pkg-config nasm
+
+        pip install --quiet --pre torch torchvision --index-url https://download.pytorch.org/whl/nightly/cu124
```

Comment:
Torchvision is required for the gpu benchmark which uses functional transforms from vision. See gpu_benchmark.py for details.

https://github.com/pytorch/torchcodec/blob/f4065f1b477148cfb0ef94167fb0bf3a63803e55/benchmarks/decoders/gpu_benchmark.py#L8

Reply:
but we don't need it to build, right? So we might prefer to install it later. This is OK though, since this job is just a temporary setup I guess.

Reply:
That's a good point. Should I move it below after the build step so it doesn't accidentally leak as a build-time dep? WDYT?